opencxl/drivers/cxl_bus_driver.py

```diff
@@ -0,0 +1,442 @@
+            }
+
+            if dvsec_id not in dvsec_function_map:
+                continue
+
+            self._create_message(f"Found DVSEC - {CXL_DVSEC_ID(dvsec_id).name}")
+            dvsec_function = dvsec_function_map[dvsec_id]
+            if not dvsec_function:
+                continue
+
+            await dvsec_function(device_info)
+
+    # pylint: enable=duplicate-code
+
+    async def _scan_cachemem_registers(self, device_info: CxlDeviceInfo):
+        component_register_info = device_info.get_register_by_type(CXL_REGISTER_TYPE.COMPONENT)
+        if not component_register_info:
+            return
+
+        cxl_cachemem_offset = component_register_info.address + 0x1000
+
+        logger.info(
+            self._create_message(
+                f"Scanning CXL.cache and CXL.mem Registers at 0x{cxl_cachemem_offset:x}"
+            )
+        )
+
+        cxl_capability_header = await self._root_complex.read_mmio(
+            cxl_cachemem_offset, CXL_CACHEMEM_REGISTER_HEADER_SIZE
+        )
+        # TODO: Define constants for masks
+        cxl_capability_id = cxl_capability_header & 0xFFFF
+        cxl_capability_version = (cxl_capability_header >> 16) & 0xF
+        cxl_cachemem_version = (cxl_capability_header >> 20) & 0xF
+        array_size = (cxl_capability_header >> 24) & 0xFF
+
+        logger.info(self._create_message(f"cxl_capability_id: {cxl_capability_id:x}"))
+        logger.info(self._create_message(f"cxl_capability_version: {cxl_capability_version:x}"))
+        logger.info(self._create_message(f"cxl_cachemem_version: {cxl_cachemem_version:x}"))
+        logger.info(self._create_message(f"array_size: {array_size:x}"))
+
+        if cxl_capability_id != CXL_CACHEMEM_REGISTER_CAPABILITY_ID.CXL:
+            return
+
+        for header_index in range(array_size):
+            header_offset = (
+                header_index + 1
+            ) * CXL_CACHEMEM_REGISTER_HEADER_SIZE + cxl_cachemem_offset
+            header_info = await self._root_complex.read_mmio(
+                header_offset, CXL_CACHEMEM_REGISTER_HEADER_SIZE
+            )
+
+            # TODO: Define constants for masks
+            cxl_capability_id = header_info & 0xFFFF
+            cxl_capability_version = (header_info >> 16) & 0xF
+            offset = (header_info >> 20) & 0xFFF
+            cxl_capability_address = cxl_cachemem_offset + offset
+            capability_name = CXL_CACHEMEM_REGISTER_CAPABILITY_ID(cxl_capability_id).name
+            logger.info(self._create_message(f"Found {capability_name} Capability Header"))
+            device_info.cachemem_registers[cxl_capability_id] = CxlCacheMemRegisterInfo(
+                id=cxl_capability_id,
+                version=cxl_capability_version,
+                offset=offset,
+                address=cxl_capability_address,
+            )
+
+    async def _scan_component_register(self, device_info: CxlDeviceInfo):
+        await self._scan_cachemem_registers(device_info)
```

Comment:
Why not call directly `self._scan_cachemem_registers()`? Will there be more on this function?

Reply:
Yup. There will be more. CacheMem register is a subset of Component Register.